internal/event/machineevent/machineevent_test.go

```diff
@@ -0,0 +1,148 @@
+			err := es.AddEvent(apiMetadata, eventType, reason, message)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(es.ListEvents()).To(HaveLen(1))
+		})
+
+		It("should handle error when retrieving metadata", func() {
+			badMetadata := api.Metadata{
+				ID: "test-id-1234"}
+			err := es.AddEvent(badMetadata, eventType, reason, message)
+			Expect(err).To(HaveOccurred())
+			Expect(es.ListEvents()).To(HaveLen(0))
+		})
+
+		It("should override the oldest event when the store is full", func() {
+			for i := 0; i < maxEvents; i++ {
+				err := es.AddEvent(apiMetadata, eventType, reason, fmt.Sprintf("%s %d", message, i))
+				Expect(err).ToNot(HaveOccurred())
+				Expect(es.ListEvents()).To(HaveLen(i + 1))
+			}
+
+			err := es.AddEvent(apiMetadata, eventType, reason, "New Event")
+			Expect(err).ToNot(HaveOccurred())
+
+			events := es.ListEvents()
+			Expect(events).To(HaveLen(maxEvents))
+			Expect(events[maxEvents-1].Spec.Message).To(Equal("New Event"))
+		})
+	})
+
+	Context("RemoveExpiredEvents", func() {
+		It("should remove events whose TTL has expired", func() {
+			err := es.AddEvent(apiMetadata, eventType, reason, message)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(es.ListEvents()).To(HaveLen(1))
+
+			Eventually(func(g Gomega) bool {
+				es.RemoveExpiredEvents()
+				return g.Expect(es.ListEvents()).To(HaveLen(0))
+			}, eventTTL+1*time.Second, 100*time.Millisecond).Should(BeTrue())
+		})
+
+		It("should not remove events whose TTL has not expired", func() {
+			err := es.AddEvent(apiMetadata, eventType, reason, message)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(es.ListEvents()).To(HaveLen(1))
+
+			es.RemoveExpiredEvents()
+			Expect(es.ListEvents()).To(HaveLen(1))
+		})
+	})
+
+	Context("Start", func() {
+		It("should periodically remove expired events", func() {
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+
+			resyncInterval := 2 * time.Second
+			go es.Start(ctx, log, resyncInterval)
+
+			err := es.AddEvent(apiMetadata, eventType, reason, message)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(es.ListEvents()).To(HaveLen(1))
+
+			Eventually(func(g Gomega) bool {
+				return g.Expect(es.ListEvents()).To(HaveLen(0))
+			}, resyncInterval+1*time.Second, 100*time.Millisecond).Should(BeTrue())
+		})
+	})
+
+	Context("ListEvents", func() {
+		It("should return all current events", func() {
+			err := es.AddEvent(apiMetadata, eventType, reason, message)
+			Expect(err).ToNot(HaveOccurred())
+
+			events := es.ListEvents()
+			Expect(events).To(HaveLen(1))
+			Expect(events[0].Spec.Message).To(Equal(message))
+		})
+
+		It("should return a copy of events", func() {
+			err := es.AddEvent(apiMetadata, eventType, reason, message)
+			Expect(err).ToNot(HaveOccurred())
+			events := es.ListEvents()
+			Expect(events).To(HaveLen(1))
+
+			events[0].Spec.Message = "Changed Message"
+
+			storedEvents := es.ListEvents()
+			Expect(storedEvents[0].Spec.Message).To(Equal(message))
```

Comment:
this was little worse readable for me. Maybe more straightforward will be
Expect(storeEvents[0].Spec.Message).ToNot(Equal(events[0].Spec.Message))

it is just cosmetic changes but i couldn't easy connect change message in events with this final comparison.

Reply:
changed.